WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift

```diff
@@ -984,18 +997,34 @@ import WordPressShared
     @objc func handleRefresh(_ sender: UIRefreshControl) {
         if !canSync() {
             cleanupAfterSync()
-            if !connectionAvailable() {
-                _ = DispatchDelayedAction(delay: .seconds(1)) {
-                    let title = NSLocalizedString("Unable to Sync", comment: "Title of error prompt shown when a sync the user initiated fails.")
-                    let message = NSLocalizedString("The Internet connection appears to be offline.", comment: "Message of error prompt shown when a sync the user initiated fails.")
-                    WPError.showAlert(withTitle: title, message: message)
-                }
-            }
+            handleConnectionError()
             return
         }
         syncHelper.syncContentWithUserInteraction(true)
     }
 
+    private func handleConnectionError() {
+        if shouldPresentAlert() {
+            _ = DispatchDelayedAction(delay: .milliseconds(500)) {[weak self] in
+                self?.presentNoNetworkAlert()
+            }
+        }
+    }
+
+    private func shouldPresentAlert() -> Bool {
+        return connectionAvailable() == false && isTableViewEmpty() == false
```

Comment:
Personally here I would prefer !connectionAvailable() && !isTableViewEmpty()

Reply:
I do have a very hard time reading negative logic, that's why I followed that approach. I'm updating though ;)